Source/WTF/wtf/glib/GMallocString.h

```diff
@@ -0,0 +1,127 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/Compiler.h>
+#include <wtf/HashFunctions.h>
+#include <wtf/glib/GSpanExtras.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/CStringView.h>
+#include <wtf/text/StringCommon.h>
+#include <wtf/text/SuperFastHash.h>
+
+namespace WTF {
+
+class PrintStream;
+
+class GMallocString final {
+    WTF_FORBID_HEAP_ALLOCATION;
+public:
+
+    static GMallocString adoptUnsafeFromUTF8(char* string)
+    {
+        if (!string)
+            return GMallocString();
+        return GMallocString { adoptGMallocSpan<char8_t, GMalloc>(unsafeMakeSpan(byteCast<char8_t>(string), string ? std::char_traits<char>::length(string) + 1 : 0)) };
+    }
+
+    static GMallocString adoptUnsafeFromUTF8(GUniquePtr<char>&& pointer)
+    {
+        return adoptUnsafeFromUTF8(pointer.release());
+    }
+
+    static GMallocString adoptUnsafeFromUTF8(GUniqueOutPtr<char>&& pointer)
+    {
+        return adoptUnsafeFromUTF8(pointer.release());
+    }
+
+    static GMallocString adoptFromUTF8(std::span<char> string)
+    {
+        if (string.size() < 1)
+            return GMallocString();
+        RELEASE_ASSERT(string[string.size() - 1] == '\0');
+        return GMallocString { adoptGMallocSpan<char8_t, GMalloc>(byteCast<char8_t>(string)) };
+    }
+
+    explicit GMallocString(const CStringView& view)
+    {
+        m_spanWithNullTerminator = dupGMallocSpan(view.spanIncludingNullTerminator());
+    }
+
+    WTF_EXPORT_PRIVATE void dump(PrintStream& out) const;
+
+    GMallocString() = default;
+    constexpr GMallocString(std::nullptr_t)
+        : GMallocString()
+    { }
+
+    bool isNull() const { return m_spanWithNullTerminator.span().empty(); }
+    const char* utf8() const LIFETIME_BOUND { return byteCast<char>(m_spanWithNullTerminator.span().data()); }
+    char* leakUTF8() WARN_UNUSED_RETURN { return byteCast<char>(m_spanWithNullTerminator.leakSpan().data()); }
+    size_t lengthInBytes() const { return !m_spanWithNullTerminator.span().empty() ? m_spanWithNullTerminator.span().size() - 1 : 0; }
+    std::span<const char8_t> span() const LIFETIME_BOUND { return m_spanWithNullTerminator.span().first(lengthInBytes()); }
+    std::span<const char8_t> spanIncludingNullTerminator() const LIFETIME_BOUND { return m_spanWithNullTerminator.span(); }
+    size_t isEmpty() const { return m_spanWithNullTerminator.span().size() <= 1; }
+
+    explicit operator bool() const { return !isEmpty(); }
+    bool operator!() const { return isEmpty(); }
+
+private:
+    explicit GMallocString(GMallocSpan<char8_t>&& string)
+        : m_spanWithNullTerminator(WTFMove(string))
+    {
+    }
+
+    GMallocSpan<char8_t> m_spanWithNullTerminator;
+};
+
+inline bool operator==(const GMallocString& a, const GMallocString& b)
+{
+    return equal(a.span(), b.span());
+}
+
+inline bool operator==(const GMallocString& a, ASCIILiteral b)
```

Comment:
Any reason why the second parameter cannot be `const&`? Like this:


```suggestion
inline bool operator==(const GMallocString& a, const ASCIILiteral& b)
```

Reply:
ASCIILiteral and CStringView are very small classes and doing that is not a big deal. GMallocString is different because it can't be copied.

Reply:
That's reasonable 👍🏼